lib/internal/test_runner/test.js

```diff
@@ -195,6 +209,18 @@ class Test extends AsyncResource {
       this.timeout = timeout;
     }
 
+    if (testNamePatterns !== null) {
+      // eslint-disable-next-line no-use-before-define
+      const match = this instanceof TestHook || ArrayPrototypeSome(
+        testNamePatterns,
+        (re) => RegExpPrototypeExec(re, name) !== null
+      );
+
+      if (!match) {
+        skip = 'test name does not match pattern';
+      }
+    }
+
     if (testOnlyFlag && !this.only) {
```

Comment:
do we allow using `--test-only` in combination with `--test-name-pattern`? what happens if they conflict?

Reply:
You can combine the two flags. I tested it locally, but I'll add a regression test for it.